test/integration/singlecluster/scheduler/fairsharing/fair_sharing_test.go

```diff
@@ -649,6 +649,260 @@ var _ = ginkgo.Describe("Scheduler", ginkgo.Ordered, ginkgo.ContinueOnFailure, f
+				cq2 *kueue.ClusterQueue
+			)
+			ginkgo.BeforeEach(func() {
+				fungibility := kueue.FlavorFungibility{
+					WhenCanBorrow:  kueue.TryNextFlavor,
+					WhenCanPreempt: kueue.TryNextFlavor,
+				}
+				preemption := kueue.ClusterQueuePreemption{
+					ReclaimWithinCohort: kueue.PreemptionPolicyAny,
+					WithinClusterQueue:  kueue.PreemptionPolicyLowerPriority,
+				}
+
+				cq1 = createQueue(testing.MakeClusterQueue("cq1").
+					Cohort("root").
+					FairWeight(resource.MustParse("1")).
+					ResourceGroup(
+						*testing.MakeFlavorQuotas("flavor1").Resource(corev1.ResourceCPU, "3").Obj(),
+					).
+					FlavorFungibility(fungibility).
+					Preemption(preemption).
+					Obj())
+
+				cq2 = createQueue(testing.MakeClusterQueue("cq2").
+					Cohort("root").
+					FairWeight(resource.MustParse("1")).
+					ResourceGroup(
+						*testing.MakeFlavorQuotas("flavor1").Resource(corev1.ResourceCPU, "0").Obj(),
+					).
+					FlavorFungibility(fungibility).
+					Preemption(preemption).
+					Obj())
+			})
+
+			ginkgo.It("workload which fits behind ClusterQueue head is able to admit", func() {
+				ginkgo.By("Creating borrowing workloads in queue2")
+				createWorkload("cq2", "1")
+				createWorkload("cq2", "1")
+				util.ExpectReservingActiveWorkloadsMetric(cq2, 2)
+
+				ginkgo.By("Create inadmissible workload in queue2")
+				createWorkloadWithPriority("cq1", "4", 999)
+
+				ginkgo.By("Verify doesn't admit")
+				util.ExpectReservingActiveWorkloadsMetric(cq1, 0)
+
+				ginkgo.By("Create admissible workload in queue2")
+				createWorkloadWithPriority("cq1", "3", 0)
+
+				ginkgo.By("Complete preemption")
+				util.FinishEvictionOfWorkloadsInCQ(ctx, k8sClient, cq2, 2)
+
+				ginkgo.By("Expect active workloads")
+				util.ExpectReservingActiveWorkloadsMetric(cq1, 1)
+				util.ExpectReservingActiveWorkloadsMetric(cq2, 0)
+				util.ExpectClusterQueueWeightedShareMetric(cq1, 0)
+				util.ExpectClusterQueueWeightedShareMetric(cq2, 0)
+			})
+
+			ginkgo.It("sticky workload deleted, next workload can admit", func() {
+				ginkgo.By("Creating borrowing workloads in queue2")
+				createWorkload("cq2", "1")
+				createWorkload("cq2", "1")
+				util.ExpectReservingActiveWorkloadsMetric(cq2, 2)
+
+				ginkgo.By("Create inadmissible workload in queue2")
+				createWorkloadWithPriority("cq1", "4", 999)
+
+				ginkgo.By("Verify doesn't admit")
+				util.ExpectReservingActiveWorkloadsMetric(cq1, 0)
+
+				ginkgo.By("Create admissible workloads in queue2")
+				stickyWorkload := createWorkloadWithPriority("cq1", "3", 99)
+
+				ginkgo.By("Another admissible workload in queue2")
+				createWorkloadWithPriority("cq1", "3", 0)
+
+				ginkgo.By("Delete sticky workload")
+				util.ExpectObjectToBeDeleted(ctx, k8sClient, stickyWorkload, true)
+
+				ginkgo.By("Complete preemption")
+				util.FinishEvictionOfWorkloadsInCQ(ctx, k8sClient, cq2, 2)
+
+				ginkgo.By("Expect active workloads")
+				util.ExpectReservingActiveWorkloadsMetric(cq1, 1)
```

Comment:
I tried running locally and hit a flake here on the first run:

```
  [FAILED] Timed out after 10.000s.
  The function passed to Eventually failed at /usr/local/google/home/michalwozniak/go/src/sigs.k8s.io/kueue/test/util/util.go:571 with:
  Expected
      <int>: 0
  to equal
      <int>: 1
  In [It] at: /usr/local/google/home/michalwozniak/go/src/sigs.k8s.io/kueue/test/integration/singlecluster/scheduler/fairsharing/fair_sharing_test.go:898 @ 10/03/25 17:11:51.661
```
PTAL

Reply:
I see, maybe this is fixed by the metric change? I will also test locally with the metric change

Reply:
Indeed, I tested locally 60 times after the latest changes and no failures, feel free to resolve.

Reply:
It flaked 3/128 times for me.

Reply:
There is a race condition where the preempted workloads schedule, given that we consider the inadmissible head after deleting the sticky workload